online-shop/src/app/add-product/add-product.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { ActivatedRoute, Router } from '@angular/router';
+import { DataService } from '../data.service';
+
+@Component({
+  selector: 'app-add-product',
+  templateUrl: './add-product.component.html',
+  styleUrls: ['./add-product.component.scss']
+})
+export class AddProductComponent implements OnInit {
+  addForm = new FormGroup({
+    name: new FormControl('', Validators.required),
+    category: new FormControl('', Validators.required),
+    image: new FormControl('', Validators.compose([
+      Validators.required,
+      this.validateImage
+    ])),
+    price: new FormControl(0.01, Validators.compose([
+      Validators.required,
+      this.validatePrice
+    ])),
+    description: new FormControl('', Validators.required),
+  })
+  constructor(private dataService: DataService, private route: ActivatedRoute, private router: Router) { }
+
+  onSubmit() {
+    try {
+      this.dataService.addProduct(this.addForm.value)
+        .subscribe(data => {
+          this.router.navigateByUrl("/products");
+          console.log(data);
```

Comment:
Never let console.log in the code. It should be used only for testing/debugging purposes